### ISSUE TITLE
Yielding: attempt to disconnect signal 'idle_frame' while emitting (locks: 1)

### ISSUE THREAD

**Issue by Xrayez** [bitwes/Gut#136]
Gut version: 6.8.0
Godot version: godotengine/godot@22637beb2 (3.2.beta)

## Problem

```gdscript
yield(yield_for(1.0, 'Waiting for the mouse to align'), YIELD)
## .. test code
yield(yield_to(get_tree(), 'idle_frame', 1.0), YIELD)
```

```log
    /# Yielding (1s):  Waiting for the mouse to align #/
    /# Yielding to signal "idle_frame" or for 1 seconds #/
ERROR: Attempt to disconnect signal 'idle_frame' while emitting (locks: 1).
   At: core\object.cpp:1522
```

Tests just hang waiting for the signal (the timer counts).

Caused by godotengine/godot@22637beb2 (which makes sense I guess)

## Solution
Might be a workaround but I changed disconnecting to be deferred which fixes the issue:

```gdscript
func _yielding_callback(from_obj=false):
	if(_yielding_to.obj):
		_yielding_to.obj.call_deferred("disconnect" ,_yielding_to.signal_name, self, '_yielding_callback')
```

**Comment by bitwes** [comment]
That makes sense.  Good find.  I'll add that into the next release which
will come out with Godot 3.2.

**Comment by bitwes** [comment]
Implemented in 6.8.1
